src/tk/widget/imagearray.c

```diff
@@ -421,22 +423,27 @@ static int iar_mclick( Widget* iar, int button, int x, int y )
  *
  *    @param iar Widget receiving the event.
  *    @return 1 if event is used.
+ *    @TODO It would be more precise to record the iar_focus result for
+ *          every click, so rapidly clicking an icon, out of bounds, and
+ *          the same icon wouldn't register as a double-click.
  */
 static int iar_mdoubleclick( Widget* iar, int button, int x, int y )
 {
    /* Handle different mouse clicks. */
+   iar_setAltTextPos( iar, x, y );
    switch (button) {
       case SDL_BUTTON_LEFT:
-         iar_focus( iar, x, y );
-         if (iar->dat.iar.dblptr != NULL)
+         if (iar->dat.iar.dblptr != NULL
+             && iar->dat.iar.selected >= 0
+             && iar->dat.iar.selected == iar_focusImage( iar, x, y )) {
```

Comment:
wouldn't it make more sense to also add a motion check in the toolkit itself so if the double click is done with two clicks far away from each other it isn't a double click?